pkg/controller/controller.go

```diff
@@ -371,6 +373,13 @@ func (c *Controller) processManagedCluster(cluster interface{}) {
 			return
 		}
 
+		// Sync the cache so that resources (like secrets) are in the cache
```

Comment:
does this code need to be moved?

Reply:
Yes, that was the root of the problem, we were processing beats resources before the cache was synced